tokio-trace-tower-http/src/lib.rs

```diff
@@ -25,10 +25,13 @@ impl<T> InstrumentedHttpService<T> {
         Self { inner, span: None }
     }
 
-    pub fn with_span(inner: T, span: Span) -> Self {
+    pub fn with_span<S>(inner: T, span: S) -> Self
+    where
+        S: Into<Option<Span>>,
```

Comment:
TIOLI: Since this isn't a trait method, I think instead of a generic parameter you could just use `span: impl Into<Option<Span>>`, which may be more concise?